pkg/graph/constructor.go

```diff
@@ -61,20 +296,20 @@ func (g *Graph) AddChannel(channel messagingv1.Channel) {
 	}
 	dest := &duckv1.Destination{Ref: ref}
 
-	v := g.getOrCreateVertex(dest)
+	v := g.getOrCreateVertex(dest, channel)
 
 	if channel.Spec.Delivery == nil || channel.Spec.Delivery.DeadLetterSink == nil {
 		// no DLS, we are done
 		return
 	}
 
 	// channel has a DLS, we need to add an edge to that
-	to := g.getOrCreateVertex(channel.Spec.Delivery.DeadLetterSink)
+	to := g.getOrCreateVertex(channel.Spec.Delivery.DeadLetterSink, nil)
 
 	v.AddEdge(to, dest, NoTransform{}, true)
 }
 
-func (g *Graph) AddEventType(et *eventingv1beta3.EventType) error {
+func (g *Graph) AddEventType(et eventingv1beta3.EventType) error {
 	ref := &duckv1.KReference{
 		Name:       et.Name,
 		Namespace:  et.Namespace,
```

Comment:
Below this line, we have a `APIVersion: "eventing.knative.dev/v1beta3"`.

This `v1beta3` information is probably needed for this case for using as a map key. Group (`eventing.k.d`) should be enough, as there can't be another resource with the same name and group, but in a different version.

Ignoring the version can make things easier. Would we lose important information to be provided to the clients of this library?

Reply:
My only concern here is that if we get rid of the version, it might make it harder for clients of the library to understand what type to cast the `interface{}` value we are storing with the object to...